### ISSUE TITLE
why fixed_param_names in module have no use

### ISSUE THREAD

**Issue by lmmcc** [apache/incubator-mxnet#11897]
i want to fixed all the layer's parameter  befor 'fc7'layer,and i change the code like this:

`name_list=[name for name in sym.list_arguments() if 'fc7' not in name]
    print(name_list)
    #label_name = 'softmax_label'
    #label_shape = (args.batch_size,)
    model = mx.mod.Module(
        context       = ctx,
        symbol        = sym,
        #fixed_param_names=name_list
    )`
but when i run using pretrained model ,and save a checkpoint after one epoch,,,i compare the model and the pretrained model,,,i find that the layers' parameters before 'fc7'layer also changed,,,i don't know why,,,can anyone please help me.

**Comment by wkcn** [comment]
I think you can try to fix the layers' parameters before 'fc7' layer.
fc7 = sym.get_internals()['fc7_output']
name_list = fc7.list_arguments()

**Comment by sandeep-krishnamurthy** [comment]
@lmmcc Thanks for using MXNet. For how to and usability questions, you can use MXNet discuss forum - https://discuss.mxnet.io/ where there are more people who can answer your query. We want to use Github issues mainly for bugs and feature request.

Closing the issue here. Please reopen if you are finding any issues.

**Comment by lmmcc** [comment]
ok,thanks very much

**Comment by wkcn** [comment]
It seems that the parameters' name in *.param files may not match that of the models.

When you fine-tune the model or check the difference between *.param files, the parameters are re-initiallized. I remember that the random seed is fixed in MXNet, so the parameters are the same.

You can try to add the parameter `allow_missing = False` in `model.fit`

**Comment by lmmcc** [comment]
@wkcn  if I use the `allow_missing = False in model.fit` how can i change the 'fc7' layers' class number....which i want to do is using a small number of training data to finetune the pre-trained model which is good enough to extract feature.

**Comment by Achhhe** [comment]
hey,did you deal with the problem? I want to fix the params before fc1  that are used to extract features, and only train the new fc layers. I used two methods , fixed_param_names and  sgd.set_lr_mult, but both didnt work @wkcn @lmmcc

**Comment by wkcn** [comment]
@Achhhe Hi. [Issue #9511](https://github.com/apache/incubator-mxnet/issues/9511) may be helpful.
You need to set the variable `idx2name`.

**Comment by Achhhe** [comment]
@wkcn thx for help, I have delt with the problem by fixed_param_names. The reason why the method didn't work is momentum=0.9 in BN, after I set the momentum=1 in BN, the params are fixed
